templates/page.html

```diff
@@ -1,5 +1,12 @@
+{% import "macros/head.html" as head -%}
 {% extends "index.html" -%}
 
+{%- block math -%}
+  {% if page.extra.math %}
+  {{ head::katex() }}
+  {% endif %}
```

Comment:
@ejmg There's two levels of checks that are preventing this katex addition from adding to total download size for anyone who doesn't explicitly turn on katex support.

1. This is turned on a per-page level, meaning that even when it's enabled for a whole site, it will only add to the total download size for the specific pages that enable it.
2. The check on line 46 of `head.html` enables/disables the pertinent resources across the entire site *even if* a specific page attempted to turn it on.

In short, this should be a pretty safe change re: download size, and won't impact other users of the theme, aside from add unused files to their `static/assets` directory and taking up (minimal) room on their server.

Reply:
I think it's better leaving per-page level control instead of the global one. Now the math works by default on every article with `math=true` in the frontmatter (`[extra]` section, of course). Other articles aren't touched in any way from this feature.